File: frontend/src/types/message.ts
export interface Message {
  id: string;
  chat_id: string;
  user_id: string;
  content: string | null;
  type: string;
  created_at: string;
  attachments?: { id: string; url: string; type?: string; filename?: string }[];
}

export interface SendMessagePayload {
  chatId: string;
  content: string;
  type?: string;
  tempId?: string;
}

File: backend-fastapi/app/schemas/message.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel


class MessageBase(BaseModel):
    content: str | None = None
    type: str = "text"


class AttachmentCreate(BaseModel):
    url: str
    type: str | None = None
    filename: str | None = None


class MessageCreate(MessageBase):
    attachments: list[AttachmentCreate] = []


class MessageUpdate(BaseModel):
    content: str | None = None


class AttachmentResponse(BaseModel):
    id: UUID
    url: str
    type: str | None
    filename: str | None

    class Config:
        from_attributes = True


class MessageResponse(BaseModel):
    id: UUID
    chat_id: UUID
    user_id: UUID
    content: str | None
    type: str
    created_at: datetime
    updated_at: datetime | None = None
    attachments: list[AttachmentResponse] = []
    sender_name: str | None = None  # для групповых чатов — ник отправителя

    class Config:
        from_attributes = True
